Add execution time for each line / root method

nkns165 opened the pull request:
I added executionTime field to StackLine and RootMethodRunResult
so as to display execution time for whole method and each code line in test class in the report.
Execution time is measured in HookMethod Manager using system time.

Although some test data is modified to this function,
I didn't add / modify any test method.

nkns165 committed:
Add execution time for each code line

nkns165 committed:
Fix unique key problem for code line
- Combine method name and line number for uniqueness

NozomiIto commented:
This looks good feature! Thanks for your contribution.
Before I merge this pull request, I want you to modify several point.

### ExecutionTime specification

What is the specification of new key "executionTime" in [RootMethodRunResult](https://github.com/SahaginOrg/sahagin-java/wiki/RunResults#root-method-run-result-file) and  [StackLine](https://github.com/SahaginOrg/sahagin-java/wiki/RunResults#stackline-definition)?

According to the added logics of HookMethodManager and HtmlReport in this pull request,
- ExecutionTime of RootMethodRunResult set by HookMethodManager seems to be always zero.
- ExecutionTime of StackLine set by HookMethodManager seems to be execution time of the line if the line does not have any sub line, but not the line execution time if the line has sub line.

Since RunResults specification is very important (because this is the interface between various Sahagin modules and language bindings), and I want to keep this specification clear and consistent.

I think "executionTime" in RootMethodRunResult should represent total root method execution time, and "executionTime" in StackLine should represent total execution time of the line (including sub method execution time). And, HookMethodManager.java should generate runResults file following to the specification.

I think beforeCodeLineHook and afterCodeLineHook will make it easy to measure each line execution time, and just added these hooks from https://github.com/SahaginOrg/sahagin-java/commit/11eb3ffe5eae6342ecfe10da5ba112bc5277c4ce to https://github.com/SahaginOrg/sahagin-java/commit/4b97285c22d8cd0479f85f7c108dcfd138491ac4 (these commits have caused conflict with your pull request, sorry!!)

Could you modify the logic according to that specification?

### Sub method code lines in HTML report are broken

When root method A calls sub method B, and sub method B calls sub method C, HTML report code line is broken, since nested method call lines in HTML report are created by [sahagin-js-share](https://github.com/SahaginOrg/sahagin-js-share). Please check it.
- Note that sahagin-js-share is loaded to sahagin-java by git-submodule
- I think it is difficult for a new contributor to notice this logic.. sorry!

### Execution time column order in HTML report

When showing code in Sahagin HTML report by clicking "show code" button, page name, description, execution time, original code columns are shown in this order. But I feel order of page name, description, original code, execution time, is more appropriate. So, what do you think of it?

---

These my requests may be a little difficult, but I would appreciate if you could improve your pull request.
Please feel free to ask me any question about this.
I and maybe many Sahagin users are looking forward to using this new feature!

nkns165 commented:
Thank you for your comments!
I will reflect them.

nkns165 committed:
Change column order of reports (execution time is after code)

nkns165 committed:
Fix empty cell problem when submethod is nested

- There is still a problem that cells are just empty when submethod is nested,
  while they are shown as '0 ms' when there is no nest. It should be fixed in
  next modification

nkns165 committed:
Merge latest SahaginOrg changes and resolve conflict

nkns165 committed:
Move executionTime field from StackLine to LineScreenCapture

nkns165 committed:
Modify small problems about showing execution time

- Show exeuction time for nested method call
- Fix '0 ms' result to empty for not captured lines
- Remove unused lines

nkns165 committed:
Fix bug that start time for code line is overwritten during nested method call

NozomiIto commented on src/main/java/org/sahagin/runlib/runresultsgen/HookMethodManager.java:
```diff
@@ -129,6 +133,13 @@ public void afterMethodHook(String hookedClassQualifiedName, String hookedMethod
 
+        long currentTime = System.currentTimeMillis();
+        currentRunResult.setExecutionTime((int) (currentTime - startMethodTime));
+//        if (prevCodeLine != null) {
```
@nkns165 please remove this unused comment out
